math/OpenBLAS/Portfile

```diff
@@ -205,6 +205,19 @@ if {${os.platform} eq "darwin" && ${os.major} >= 23} {
     }
 }
 
+# cmake build uses CMAKE_SIZEOF_VOID_P to detect 32 or 64 bit builds,
+# handle the case of universal build by enforcing the right size
+set merger_configure_args(arm64)    -DCMAKE_SIZEOF_VOID_P=8
+set merger_configure_args(ppc)      -DCMAKE_SIZEOF_VOID_P=4
+set merger_configure_args(ppc64)    -DCMAKE_SIZEOF_VOID_P=8
+set merger_configure_args(i386)     -DCMAKE_SIZEOF_VOID_P=4
+set merger_configure_args(x86_64)   -DCMAKE_SIZEOF_VOID_P=8
+
+# in the case of non-universal build just enforce the right value
+if {!${universal_possible} || ![variant_isset universal]} {
+    configure.args-append           $merger_configure_args(${configure.build_arch})
+}
```

Comment:
I would prefer if this was done inside a darwin version check to limit it to the cases it was actually needed. i.e. it appears its really only needed on 10.6, based on the buildbots.